core/challenges.go

```diff
@@ -1,29 +1,32 @@
 package core
 
-func newChallenge(challengeType string) Challenge {
+func newChallenge(challengeType string, token string) Challenge {
```

Comment:
Can you add a super tiny ~5 line test for `newChallenge` that sets a pre-specified `token`? I think most (all?) of the existing unit tests that use `HTTPChallenge01`,`newChallenge`,etc are sending `token=""`.